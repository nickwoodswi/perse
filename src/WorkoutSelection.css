.select-workout-type{
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    width: 100%;
}
    .workout-selector{
        width: 100%;
        height: 70px;
    }
    .select-workout-type input{
        width: 100%;
        height: 65px;
        margin: 0;
    }
    .select-workout-type-toggle{
        background-color: yellow;
        display: flex;
        text-align: center;
        align-items: center;
        width: 120px;
        height: 70px;
        margin-left: 15px;
        margin-top: 5px;
    }