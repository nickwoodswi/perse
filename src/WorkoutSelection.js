import React, {Component} from 'react'
import './WorkoutSelection.css'

class WorkoutSelection extends Component {
    render() {
        if (this.props.type === 'select') {
            return(
                <>
                <div className="select-workout-type">
                    <select className="workout-selector" onChange={e => this.props.updateWorkout(e.target.value)}>
                        {this.props.selectorOptions.map((option, idx) => {
                            return(
                                <option key={idx} value={option.workouts_id}>{option.workouts_name}</option>
                            )
                        })}
                    </select>
                    OR
                    <div 
                        className="select-workout-type-toggle"
                        onClick={e => this.props.changeWorkoutType('create')}>+ADD NEW WORKOUT</div>
                </div>
                </>
            )
        }
        if (this.props.type === 'create') {
            return(
                <>
                <div className="select-workout-type">
                    <input 
                        onChange={e => this.props.define(e.target.value)}
                        id="workout-name-input" 
                        type="text" 
                        value={this.props.name} 
                        placeholder="New Workout Name" />
                    OR
                    <div 
                        className="select-workout-type-toggle"
                        onClick={e => this.props.changeWorkoutType('select')}>SELECT WORKOUT</div>
                </div>
                </>
            )
        }
    }
}

export default WorkoutSelection
