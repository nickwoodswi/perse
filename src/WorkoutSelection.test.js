import React from 'react';
import { render } from '@testing-library/react';
import WorkoutSelection from './WorkoutSelection';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders WorkoutSelection component', () => {
    render(<MemoryRouter><WorkoutSelection /></MemoryRouter>);
});

it('WorkoutSelection renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><WorkoutSelection /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});