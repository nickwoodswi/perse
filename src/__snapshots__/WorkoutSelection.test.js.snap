// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutSelection renders the UI as expected 1`] = `
<div
  className="select-workout-type"
>
  <select
    className="workout-selector"
    onChange={[Function]}
  >
    <option>
      SELECT WORKOUT
    </option>
  </select>
  OR
  <div
    className="select-workout-type-toggle"
    onClick={[Function]}
  >
    +ADD NEW WORKOUT
  </div>
</div>
`;
